quodlibet/ext/query/unique.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 LoveIsGrief
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+from quodlibet import _, print_d, print_w
+from quodlibet.plugins.query import QueryPlugin, QueryPluginError
+
+
+class UniqueQuery(QueryPlugin):
+    PLUGIN_ID = "unique_query"
+    PLUGIN_NAME = _("Unique Query")
+    PLUGIN_DESC = _("Filter search results by unique tags.")
+    key = 'unique'
+    query_syntax = _("@(unique: tag)")
+    query_description = "<tt>tag</tt> can be album, artist, title or any other tag. " \
+        "Use multiple <tt>@(unique: tag)</tt> to filter by multiple tags."
+
+    usage = f"{query_syntax}\n\n{query_description}"
+
+    def __init__(self):
+        print_d("Initialising unique query")
+        self.unique_tag_values = set()
+        """The unique tag values that have been seen in the songs being filtered"""
+        self._reported = set()
+        """Unique errors to counter error log spam"""
+
+    def search(self, data, body):
```

Comment:
Could you add some type annotations? Trying to do that for all / nearly all new code

Reply:
I added it for the method signature. Dunno if variables and such also need them. If would be to have `mypy` check them and throw errors for where you want type hints to be put.